core/vm/concrete_bindings_test.go

```diff
@@ -0,0 +1,215 @@
+package vm
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+func newBlockContext() BlockContext {
+	var (
+		block0Hash = crypto.Keccak256Hash([]byte("block0"))
+		block1Hash = crypto.Keccak256Hash([]byte("block1"))
+		randomHash = crypto.Keccak256Hash([]byte("random"))
+	)
+
+	getHash := func(blockNumber uint64) common.Hash {
+		switch blockNumber {
+		case 0:
+			return block0Hash
+		case 1:
+			return block1Hash
+		}
+		return common.Hash{}
+	}
+
+	blockCtx := BlockContext{
+		CanTransfer: func(StateDB, common.Address, *uint256.Int) bool { return true },
+		Transfer:    func(StateDB, common.Address, common.Address, *uint256.Int) {},
```

Comment:
@therealbytes I leave this one cause is needed for .call(), .create() and create2()